common/cuda_hip/matrix/batch_dense_kernels.hpp.inc

```diff
@@ -211,6 +211,48 @@ __global__ __launch_bounds__(default_block_size, sm_multiplier) void add_scaled(
 }
 
 
+template <typename ValueType>
+__device__ __forceinline__ void add_scale(
+    const gko::batch_dense::BatchEntry<const ValueType>& alpha,
+    const gko::batch_dense::BatchEntry<const ValueType>& x,
+    const gko::batch_dense::BatchEntry<const ValueType>& beta,
+    const gko::batch_dense::BatchEntry<ValueType>& y)
+{
+    const int max_li = x.num_rows * x.num_rhs;
+    for (int li = threadIdx.x; li < max_li; li += blockDim.x) {
+        const int row = li / x.num_rhs;
+        const int col = li % x.num_rhs;
+
+        if (alpha.num_rhs == 1) {
```

Comment:
```suggestion
        if (alpha.num_rhs == 1 && beta.num_rhs==1) {
```

Reply:
This is already being checked in core.

Reply:
IMO, this makes the code more readable and clear, but it is not that important.